src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Services/EFormidlingAccessValidator.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Threading.Tasks;
+
+using Altinn.Common.AccessToken.Services;
+using Altinn.Platform.Authentication.Model;
+using Altinn.Platform.Authentication.Services.Interfaces;
+
+using Microsoft.IdentityModel.Tokens;
+
+namespace Altinn.Platform.Authentication.Services
+{
+    /// <summary>
+    /// Implementation of an eFormidling access validator
+    /// </summary>
+    public class EFormidlingAccessValidator : IEFormidlingAccessValidator
+    {
+        private readonly IAccessTokenValidator _validator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EFormidlingAccessValidator"/> class.
+        /// </summary>
+        public EFormidlingAccessValidator(IAccessTokenValidator validator)
+        {
+            _validator = validator;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IntrospectionResponse> ValidateToken(string token)
+        {
+            IntrospectionResponse result = new();
+
+            bool isValid = await _validator.Validate(token);
+
+            if (isValid)
+            {
+                JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
+                JwtSecurityToken jwt = tokenHandler.ReadJwtToken(token);
+
+                result.Active = isValid;
+                result.Iss = jwt.Issuer;
+            }
+
+            return result;
```

Comment:
Would it be an idea to set Iss also when the token is invalid?

Reply:
I was wondering the same, but decided on not. The reason being we would have to first ensure that we are able to read the token before parsing it into a jwtSecurityToken to avoid an exception. If the token is valid, this has already been asserted in the underlying service. 

Iss is defined as optional by the standard, and in our case, the issuer elsewhere in the solution when the token is invalid.

Reply:
I think my primary concern is that we aren't explicit about values when the validation fails. While at the same time it's a good thing that the default values indicate failiure. Leave it as is.